openfisca_core/tracers.py

```diff
@@ -108,11 +112,30 @@ def record_calculation_start(self, variable_name, period, **parameters):
             self.requested_calculations.add(key)
 
         if not self.trace.get(key):
-            self.trace[key] = {'dependencies': []}
+            self.trace[key] = {'dependencies': [], 'parameters': {}}
+
         self.stack.append(key)
         self._computation_log.append((key, len(self.stack)))
         self.usage_stats[variable_name]['nb_requests'] += 1
 
+    def record_calculation_parameter_access(self, parameter, period):
+        at_instant = parameter.get_at_instant(period)
+
+        if isinstance(at_instant, AbstractTaxScale):
+            values = at_instant.amounts if isinstance(at_instant, AmountTaxScale) else at_instant.rates
+            at_instant_trace = {}
```

Comment:
Defining a dict by setting the values one by one imperatively is very old school and doesn't read well. Prefer a comprehensive definition like:

```py
        {
            str(threshold): values[index]
            for index, threshold in enumerate(self.thresholds)
            }
```

Reply:
Done